source/core/DualTypeVisitor.cpp

```diff
@@ -60,6 +60,9 @@ namespace Vireo
                 }
                 break;
             }
+            case kEncoding_None:
+                success = typeRefY->BitEncoding() == kEncoding_None;
```

Comment:
I will add separate encoding for variant as part of a fix for a CAR. However, until then, I think we should check for IsVariant() on both types to find out if both of them are variant. 

Otherwise, just leave a TODO that kEncoding_None check exists only for checking variants (if it is) and that this should be fixed with kEncoding_Variant when we add a new encoding type. When I add the new encoding type, I will likely fix IsVariant() to use the encoding than the Name() as it will be more efficient.

Reply:
The `kEncoding_None` case is only encountered when we traverse the underlying type of an empty variant constant wired through a `setattribute`. I think we will still encounter this when variants have a separate encoding. I will add a note in the code mentioning that this is for the empty variant case.